infra/README.md

```diff
@@ -20,3 +27,56 @@ To make changes to `makemake`:
       ```
       nixos-rebuild switch --show-trace -L --flake /root/ngipkgs/#makemake
       ```
+
+## Secret management
+We use [sops-nix](https://github.com/Mic92/sops-nix) to manage secrets in
+NGIpkgs. In order to access the secrets they must be encrypted with your public
+key and you must have the matching private key available in your system.
+[Sops](https://github.com/getsops/sops) must also be installed and configured.
+
+### Viewing and editing secrets
+By default on Linux, sops expects the private keys to be in
+`$XDG_CONFIG_HOME/sops/age/keys.txt` or `$HOME/.config/sops/age/keys.txt`. You
+can generate an age key using `age-keygen` or `ssh-to-age` to convert a SSH key
+to an age compatible one. Once you've setup your key, you can run `sops -d
+<file-to-decrypt>` to view an encrypted file or `sops <file-to-edit>` to edit a
+file.
+
+### Encrypting secrets for a new keυ
+To add a new person or a key edit `infra/.sops.yaml`, add the name of the
+person and their age public key under `.humans`. Additionally, you need to add
+their names to any files you want them to have access. This is done in
+`infra/.sops.yaml` under the `creation_rules` section.
+
+This is an example diff of what this might look like:
```

Comment:
Not sure I like how verbose that is. We could also refer to a commit via it's hash, or use `diff -U2` or so which would generate less lines.